app/presenters/configuration_status_presenter.rb

```diff
@@ -0,0 +1,254 @@
+class ConfigurationStatusPresenter < HostStatusPresenter
+  private
+
+  delegate :connection, to: ActiveRecord::Base
+  delegate :exec_query, to: :connection
+
+  def total_data
+    @total_data ||= exec_query(total_data_query.to_sql)
+      .rows
+      .each_with_object({}) { |(status, count), memo| memo[status] = count }
+  end
+
+  def owned_data
+    @owned_data ||= exec_query(owned_data_query.to_sql)
+      .rows
+      .each_with_object({}) { |(status, count), memo| memo[status] = count }
+  end
+
+  def total_data_query
+    host_status_table
+      .where(host_status_table[:type].eq('HostStatus::ConfigurationStatus')
+        .and(is_relevant))
+      .join(hosts_table)
+        .on(host_status_table[:host_id].eq(hosts_table[:id])
+          .and(hosts_table[:type].eq('Host::Managed')))
+      .project(computed_status)
+      .project(arel_quoted(1).count)
+      .group('computed_status')
+  end
+
+  def owned_data_query
+    host_status_table
+      .where(host_status_table[:type].eq('HostStatus::ConfigurationStatus')
+        .and(
+          host_status_table[:host_id].in(
+            Arel.sql(Host::Managed.search_for('owner = current_user').select(:id).to_sql)
+          )
+        )
+        .and(is_relevant))
+      .join(hosts_table).on(host_status_table[:host_id].eq(hosts_table[:id]).and(hosts_table[:type].eq('Host::Managed')))
+      .project(computed_status)
+      .project(arel_quoted(1).count)
+      .group('computed_status')
+  end
+
+  def is_relevant
+    hosts_table[:puppet_proxy_id].eq(nil).not
+      .or(any_reports)
+      .or(always_show_configuration_status)
+  end
+
+  def always_show_configuration_status
+    settings_table
+      .where(settings_table[:name].eq('always_show_configuration_status')
+        .and(settings_table[:value].eq('--- true')
+          .or(settings_table[:value].eq(nil).and(settings_table[:default].eq('--- true')))))
+      .exists
```

Comment:
this is terrible... why? xD
And it will not work for long as I plan to remove default from the database.
Why dont we `Setting['always_show_configuration_status']`?

Reply:
we can't simply replace it with `Setting['always_show_configuration_status']`  but `Setting['always_show_configuration_status'] ? Arel::Nodes::True.new : Arel::Nodes::False.new` should do the trick